tests/test_connector_redmine.py

```diff
@@ -0,0 +1,217 @@
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.tests.common import TransactionCase
+from openerp.tools import (DEFAULT_SERVER_DATETIME_FORMAT, DEFAULT_SERVER_DATE_FORMAT)
+from openerp.addons.connector.connector import ConnectorEnvironment
+from openerp.addons.connector.session import ConnectorSession
+from ..unit.mapper import RedmineImportMapper
+from ..unit.binder import RedmineModelBinder
+from ..unit.import_synchronizer import RedmineImportSynchronizer
+from datetime import datetime
+
+
+class TestRedmineConnector(TransactionCase):
+    def setUp(self):
+        super(TestRedmineConnector, self).setUp()
+        self.backend_model = self.env['redmine.backend']
+        self.user_model = self.env['res.users']
+        self.employee_model = self.env['hr.employee']
+        self.timesheet_model = self.env['hr.analytic.timesheet']
+        self.account_model = self.env['account.analytic.account']
+        self.redmine_model = self.env['redmine.hr.analytic.timesheet']
+
+        self.context = self.user_model.context_get(self.cr, self.uid)
+        cr, uid, context = self.cr, self.uid, self.context
+
+        self.user_id = self.user_model.create(
+            cr, uid, {
+                'name': 'User 1',
+                'login': 'user_1',
+            }, context=context)
+
+        self.user = self.user_model.browse(
+            cr, uid, self.user_id, context=context)
+
+        journal_id = self.env.ref('hr_timesheet.analytic_journal').id
+
+        self.account_id = self.account_model.create(cr, uid, {
+            'type': 'contract',
+            'name': 'Test Redmine',
+            'code': 'abcd',
+        }, context=context)
+
+        self.employee_id = self.employee_model.create(
+            cr, uid, {
+                'name': 'Employee 1',
+                'user_id': self.user_id,
+                'journal_id': journal_id,
+            }, context=context)
+
+        self.backend_id = self.backend_model.create(cr, uid, {
+            'name': 'redmine_test',
+            'location': 'http://localhost:3000',
+            'key': '39730056c1df6fb97b4fa5b9eb4bd37221ca1223',
+            'version': '1.3',
+        }, context=context)
+
+        self.backend = self.backend_model.browse(
+            cr, uid, self.backend_id, context=context)
+
+        self.session = ConnectorSession(cr, uid, context=context)
+        self.environment = ConnectorEnvironment(
+            self.backend, self.session, 'redmine.hr.analytic.timesheet')
+
+        self.now = datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
```

Comment:
For dates, use `fields.Datetime` and `fields.Date` helpers (`from_string()`, `now()`, ...)
https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.fields.Date